### ISSUE TITLE
Projection processor should support UDF with one or more arguments

### ISSUE THREAD

**Issue by arunmahadevan** [hortonworks/streamline#1217]
Support projections with one or more arguments, constants and nested invocations. Some examples

- upper(name)
- concat(first, last)
- substring(str, 0, 10)
- lower(trim(text))

One option could be to have a code text editor with pop up and use the type information available in the UDF api to prompt for the arguments.

**Comment by arunmahadevan** [comment]
@shahsank3t , you should be able to pass functions as "expr" objects within projections as follows:

```
"projections":[{"expr":"LOWER(driverName) AS DN"},{"expr":"driverId"}]
```

**Comment by arunmahadevan** [comment]
Once #1224 is merged, we could introduce similar changes for projection processor.

**Comment by sameer79** [comment]
Raised PR : https://github.com/hortonworks/streamline/pull/1255

**Comment by arunmahadevan** [comment]
thanks @sameer79 , merged #1255 to master
